File: Algo/Leetcode/012IntegerToRoman.cpp
class Solution {
public:
    string intToRoman(int num) {
        vector<int> nums = {1000,900,500,400,100,90,50,40,10,9,5,4,1};
        vector<string> romans = {"M", "CM", "D", "CD", "C", "XC", "L",
                         "XL", "X", "IX", "V", "IV", "I"};
        string res = "";
        for(int i=0; i<nums.size(); i++){
            while(num>=nums[i]){
                res += romans[i];
                num -= nums[i];
            }
        }
        return res;
    }
};

File: Algo/Leetcode/012IntegerToRoman.py
class Solution(object):
    def intToRoman(self, num):
        """
        :type num: int
        :rtype: str
        """
        d = {
            'I': 1,
            'IV': 4,
            'V': 5,
            'IX': 9,
            'X': 10,
            'XL': 40,
            'L': 50,
            'XC': 90,
            'C': 100,
            'CD': 400,
            'D': 500,
            'CM': 900,
            'M': 1000
        }
        d_l = sorted(list(d.values()))[::-1]  # 1000, 900, 500,...., 1
        _d = {}
        for k, v in d.items():
            _d[v] = k

        res = ""
        i = 0
        while num > 0:
            while num >= d_l[i]:
                res += str(_d[d_l[i]])
                num -= d_l[i]
            i += 1

        return res